brion/circuitConfig.cpp

```diff
@@ -0,0 +1,218 @@
+/* Copyright (c) 2013-2018, EPFL/Blue Brain Project
+ *                          Jonas Karlsson <jonas.karlsson@epfl.ch>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "circuitConfig.h"
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <regex>
+#include <streambuf>
+#include <string>
+
+#include "json.hpp"
+
+namespace
+{
+nlohmann::json _parseCircuitJson(const std::string& jsonStr)
+{
+    using nlohmann::json;
+
+    const auto jsonOrig = json::parse(jsonStr);
+    auto jsonFlat = jsonOrig.flatten();
+    auto manifest = jsonOrig["manifest"];
+
+    std::map<std::string, std::string> variables;
+
+    // Find variables in manifest section
+    for (auto it = manifest.begin(); it != manifest.end(); ++it)
+    {
+        const auto keyName = it.key();
+        const auto name =
+            keyName.substr(std::string(keyName).find_last_of('/') + 1);
```

Comment:
Use regex to check that variables names match this pattern "\$([A-Za-z0-9_])", throw an error if the string doesn't match